fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/planning/InExtractor.java

```diff
@@ -157,10 +193,11 @@ public void setSort(@Nonnull KeyExpression key, boolean reverse) {
             if (!found) {
                 // There is a requested sort ahead of the ones from the IN's, so we can't do it.
                 cancel();
```

Comment:
It feels a little weird that this still calls `cancel`, with the new stuff. Like, in theory, it should be able to use the extracted sort components from the `IN` and then use a `UnionPlan` for the rest. Though I think it's still _safe_ to do it as it is here, so perhaps an optimization for another time.

Reply:
Agreed, but I'd rather not mess with this code if we don't have to.